src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -61,6 +71,20 @@ public void assertIsValid(InputEnvelope envelope) {
                     ))
                     .onSuccess(res -> handleValidationResponse(res, envelope, docWithOcr))
                     .onFailure(exc -> handleRestClientException(exc, validationUrl, envelope, docWithOcr))
+                    .map(response -> ocrValidationWarnings(docWithOcr, response.warnings))
+                    .getOrElseGet(exc ->
+                        ocrValidationWarnings(
+                            docWithOcr,
+                            singletonList("OCR validation was not performed due to errors")
+                        )
+                    )
+        );
+    }
+
+    private OcrValidationWarnings ocrValidationWarnings(InputScannableItem scannableItem, List<String> warnings) {
+        return new OcrValidationWarnings(
+            scannableItem.documentControlNumber,
+            warnings != null ? warnings : emptyList()
```

Comment:
so is there a test simulating `null` and `warning = empty list` situation? perhaps just missing assertions in existing tests